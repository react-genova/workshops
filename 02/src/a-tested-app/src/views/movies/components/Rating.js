// @flow
import * as React from 'react';
import {Rating as RatingInternal} from 'semantic-ui-react';
import {withHandlers, setDisplayName} from 'recompose'

type Props = {
    rating:number, 
    maxRating:number, 
    id:string,
    onRate:(id:string, newRate:number) => void
}

//const Rating = ({rating, maxRating, onRate}:Props) => <RatingInternal rating={rating} maxRating={maxRating} onRate={(event,data)=>{onRate(data.rating)}}/>;

const onRateHandler = (props:Props) => (event:*,data:{rating:number}) => {
    props.onRate(props.id, data.rating)
};

const enhance = withHandlers({
    onRate: onRateHandler
});
const RatingInternalEnhanced = setDisplayName("Rating")(enhance(RatingInternal));
const Rating = ({rating, maxRating, onRate,id}:Props) => <RatingInternalEnhanced rating={rating} maxRating={maxRating} id={id} onRate={onRate}/>;

export {onRateHandler};
export default Rating;