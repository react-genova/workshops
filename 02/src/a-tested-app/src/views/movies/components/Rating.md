### Rating ###

```js
require('semantic-ui-css/semantic.min.css');
const onRate=(newRate)=>{alert(newRate);console.log(newRate)};
<Rating maxRating={10} rating={5} id={1} onRate={onRate}/>
```