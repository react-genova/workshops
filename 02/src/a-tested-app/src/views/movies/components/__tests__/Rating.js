//@flow
import React from 'react';
import Rating, {onRateHandler} from '../Rating';
import { shallow } from 'enzyme';

describe('Rating component', () => {

    it('Rating snapshot', () => {
        const onRate = jest.fn();
        const component = shallow(
            <Rating id="1" rating={3} maxRating={5} onRate={onRate} />
        );
        expect(component).toMatchSnapshot();
    });

    it('Rating onRate', () => {
        const onRate = jest.fn();
        const id = "1";
        const newRating = 5;
        const component = shallow(
            <Rating id={id} rating={3} maxRating={5} onRate={onRate} />
        );
        component.simulate('rate', id, newRating);
        // The mock function is called twice
        expect(onRate.mock.calls.length).toBe(1);
        // The first argument of the first call to the function was "id"
        expect(onRate.mock.calls[0][0]).toBe(id);
        // The second argument of the first call to the function was "newRating"
        expect(onRate.mock.calls[0][1]).toBe(newRating);
    });

    it('onRateHandler', () => {
        const onRate = jest.fn();
        const id = "1";
        const newRating = 5;
        const props = {
            rating: 1, 
            maxRating:5,
            id,
            onRate
        };
        const event = {};
        const data = {rating: newRating};
        onRateHandler(props)(event,data);
        // The mock function is called twice
        expect(onRate.mock.calls.length).toBe(1);
        // The first argument of the first call to the function was "id"
        expect(onRate.mock.calls[0][0]).toBe(id);
        // The second argument of the first call to the function was "newRating"
        expect(onRate.mock.calls[0][1]).toBe(newRating);
    });

});