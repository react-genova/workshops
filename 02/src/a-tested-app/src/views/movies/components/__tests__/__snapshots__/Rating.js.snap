// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Rating component Rating snapshot 1`] = `
<Rating
  id="1"
  maxRating={5}
  onRate={[Function]}
  rating={3}
/>
`;
